app/services/commitments/helper_service.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module Commitments
+  class HelperService < BaseService
+    def initialize(commitment:, invoice_subscription:, current_usage: false)
+      @commitment = commitment
+      @invoice_subscription = invoice_subscription
+
+      super
+    end
+
+    def proration_coefficient
+      result.proration_coefficient = calculate_proration_coefficient
+      result
+    end
+
+    def period_invoice_ids
+      result.period_invoice_ids = fetch_period_invoice_ids
+      result
+    end
+
+    private
+
+    attr_reader :commitment, :invoice_subscription, :current_usage
+
+    delegate :subscription, to: :invoice_subscription
+
+    def calculate_proration_coefficient
+      all_invoice_subscriptions = subscription
+        .invoice_subscriptions
+        .where(invoice_id: fetch_period_invoice_ids)
+        .order(
+          Arel.sql(
+            ActiveRecord::Base.sanitize_sql_for_conditions(
+              'COALESCE(invoice_subscriptions.to_datetime, invoice_subscriptions.created_at) ASC',
+            ),
```

Comment:
Can you please describe ordering here and what is intended? Also, I think that we should use invoice_subscription.timestamp and not created_at attribute

Reply:
yep but created_at should not be used since invoice reference is timestamp.

Reply:
I changed it to timestamp.